### ISSUE TITLE
Get-BcContainerAppInfo does not work properly with multiple published apps

### ISSUE THREAD

**Issue by MODUSCarstenScholling** [microsoft/navcontainerhelper#2050]
**Describe the issue**
Get-BcContainerAppInfo does not work properly with multiple published apps. 

App D is published two times with two different versions. In my case, two scopes, old Global, new Tenant/Dev. This is causing App D to occur multiple times in the result.

**Script used causing the issue**
```
$apps = Get-BcContainerAppInfo -containerName bc182de -tenant default -sort DependenciesLast
```

**Command result**
```
Name               Version                 Scope
----               -------                 -----
App A              18.1.21060102.21060102 Tenant
App B              17.2.21052001.21052001 Tenant
App C              18.1.21060101.21060101 Tenant
App D              17.4.21090101.0        Tenant
App D              17.0.21040701.0        Global
App E              17.1.0.16              Tenant
App F              18.1.21060101.21060101 Tenant
App D              17.4.21090101.0        Tenant
App D              17.0.21040701.0        Global
App G              17.1.1.27              Tenant
App D              17.4.21090101.0        Tenant
App D              17.0.21040701.0        Global
App H              17.2.21052702.21052702 Tenant
App D              17.4.21090101.0        Tenant
App D              17.0.21040701.0        Global
[...]
Application        18.2.26217.26490       Global
Base Application   18.2.26217.26490       Global
System Application 18.2.26217.26490       Global
```

**Suggested fix**
Replace "-eq" with "-iin".

before
```
        function AddAnApp { Param($anApp) 
            $alreadyAdded = $script:installedApps | Where-Object { $_.AppId -eq $anApp.AppId }
            if (-not ($alreadyAdded)) {
                AddDependencies -anApp $anApp
                $script:installedApps += $anApp
            }
        }
```

after
```
        function AddAnApp { Param($anApp) 
            $alreadyAdded = $script:installedApps | Where-Object { $_.AppId -iin $anApp.AppId }
            if (-not ($alreadyAdded)) {
                AddDependencies -anApp $anApp
                $script:installedApps += $anApp
            }
        }
```

**Comment by git-no** [comment]
I did had the same behavior.
My workaround was: I unpublished all duplicated not installed apps. In result `Get-BcContainerAppInfo` did work well.
In a single tenant or with within the same tenant there should no need to have the  same app (id) multiple times.

**Comment by freddydk** [comment]
@MODUSCarstenScholling do I understand this right, that the published 17.0 version of app D is published and not installed on any tenant, right?

**Comment by freddydk** [comment]
There are two issues here.
One is that the sorting only takes AppId into account - it should also include version.
While there can be more than one occurance of AppId published, there can be only one with the same appid,version.
I have added this to the equasion.

Furthermore, in a lot of cases you might only want to get apps, which are actually installed. For this, I have added the -installedOnly parameter.
